migration/sql-files/004-authorization.sql

```diff
@@ -0,0 +1,70 @@
+-- Create resource type table
+CREATE TABLE resource_type (
+    resource_type_id uuid primary key DEFAULT uuid_generate_v4(),
+    name text NOT NULL,
+    description text NULL,
+    created_at timestamp with time zone,
+    updated_at timestamp with time zone,
+    deleted_at timestamp with time zone
+);
+
+CREATE INDEX resource_type_resource_type_id_idx ON resource_type USING BTREE (resource_type_id); 
+
+CREATE TABLE resource_type_scope (
+    resource_type_scope_id uuid primary key DEFAULT uuid_generate_v4(),
+    resource_type_id uuid NOT NULL references resource_type(resource_type_id),
+    name text NOT NULL,
+    description text NULL,
+    created_at timestamp with time zone,
+    updated_at timestamp with time zone,
+    deleted_at timestamp with time zone
+);
+
+CREATE TABLE role (
+    role_id uuid primary key DEFAULT uuid_generate_v4(),
+    resource_type_id uuid NOT NULL references resource_type(resource_type_id),
+    name text NOT NULL,
+    created_at timestamp with time zone,
+    updated_at timestamp with time zone,
+    deleted_at timestamp with time zone
+);
+
+CREATE TABLE role_scope (
+    resource_type_scope_resource_type_scope_id uuid references resource_type_scope(resource_type_scope_id),
+    role_role_id uuid references role(role_id),
```

Comment:
It would be "role_id" if we use "id" as primary key in "role" table instead of "role_id", correct?

Reply:
Correct, but only because of gorm's naming convention defaulting to table_name_column_name without there being any possible way of overriding it.  This means that gorm would be totally broken for legacy databases in which it is not possible to change column names.

Reply:
Oh, and other people are experiencing this same issue:

https://stackoverflow.com/questions/33437453/using-different-join-table-columns-than-defaults-with-gorm-many2many-join

Reply:
So, another reason to use short column names for primary keys - "id"? :) I understand that's a gorm limitation but columns like resource_type_scope_resource_type_scope_id doesn't look good at all.